### DISCUSSION TITLE
Cannot import .osc file

### DISCUSSION THREAD

**Discussion by macosta541** [mtex-toolbox/mtex#840] [Ask Anything] [answered]
Hello,

I am new to both Matlab, EBSD, and Mtex, so I apologize if this is a rudimentary question (but I searched the old forum and this one and wasn't able to find a solution).

When I try to import my .osc file into Mtex using the command ebsd = EBSD.load('file_name. osc') I receive the following error :

>> EBSD.load("test.osc")
Error using check_interfaces (line 53)
Could not detect file format. You may ask
for help in the MTEX forum
(https://groups.google.com/forum/#!forum/mtexmail).
Error in EBSD.load (line 72)
  [interface,options] =
  check_interfaces(fname{1},'EBSD',varargin{:}); 

I am using Matlab R2020b and mtex 5.6.0 on windows 10.  The same error occurs when I try to use the import wizard.

Is there a way for me to convert the .osc file, which I cannot read with Textpad, into something that is more palatable for Mtex?

Many Thanks,
Marisa

**Comment by kilir** [comment]
Hi,
the .osc file format changed over time, at the moment you can read the newer ones, however the old ones do not work at the moment - something one can fix if needed. 

The best thing however is, that you export your data as .ang which is human readable.

Cheers,
Rüdiger

**Reply by macosta541** [reply]
Thank you, Rüdiger, for your prompt assistance! I will ask if I can get the data as .ang files ( I am not at the institution the data was acquired at). If I cannot, how would I go about getting the older files to work? Maybe try an older version of mtex or matlab ?

Best regards,
Marisa

**Comment by ralfHielscher** [comment]
You may send the files and we can check what caused the error.

Ralf.

********************************************************************
Ralf Hielscher                   Tel: +371-531-38556
Fakultät für Mathematik               +371-531-22200 (Sekr.)
Technische Universität Chemnitz  Fax: +371-531-22109
Reichenhainer Str. 39            E-mail:
ralf.hielscher@mathematik.tu-chemnitz.de
D-09126 Chemnitz                 http://www.tu-chemnitz.de/~rahi
********************************************************************

**Reply by macosta541** [reply]
Figuring out what is wrong with them would be good, I think.  [Here is a google drive folder with the data files](https://drive.google.com/drive/folders/1ffx5q5qw087b48uqjKR1NvMbi8tLH0-a?usp=sharing).  Thank you.

**Comment by lfgmorales** [comment]
in addition to what ralf said, I can also transform your files to *.ang files using OIM (of course, assuming you don't have 357 gigantic maps :-D)



cheers



Luiz




Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Reply by kilir** [reply]
For the sake of documentation, I could open some old osc files prior to d1e61b1a8d3b403e2ae06acda901e8d35d04c1a6
Cheers,
Rüdiger

**Reply by macosta541** [reply]
Hello Luiz,

That would be awesome, please and thank you! [Here is a google drive with the files](https://drive.google.com/drive/folders/1ffx5q5qw087b48uqjKR1NvMbi8tLH0-a?usp=sharing). The technician at the lab the data was acquired with is not confident he will be able to export the data as .ang files.

Also, I see you are at ETH - I have just started a post-doc at the University of Lausanne and will probaby be doing a good bit of EBSD + TEM work in the next three years, so perhaps I will see you around once we have all been vaccinated!

Best regards, 
Marisa

**Comment by lfgmorales** [comment, marked as answer]
Dear Maria


you will find the *.ang files here


https://polybox.ethz.ch/index.php/s/S9mu5Ij1z1foXwE

[https://polybox.ethz.ch/core/img/favicon-fb.png]<https://polybox.ethz.ch/index.php/s/S9mu5Ij1z1foXwE>

polybox<https://polybox.ethz.ch/index.php/s/S9mu5Ij1z1foXwE>
polybox.ethz.ch
Acosta is publicly shared

cheers


Luiz



[https://polybox.ethz.ch/core/img/favicon-fb.png]<https://polybox.ethz.ch/index.php/s/S9mu5Ij1z1foXwE>

polybox<https://polybox.ethz.ch/index.php/s/S9mu5Ij1z1foXwE>
polybox.ethz.ch
Acosta is publicly shared





Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Reply by ralfHielscher** [reply]
Somehow, the not working old osc files are not longer available. Could somebody upload or send them to be again.
It would be nice if you could import them.

Ralf.

**Reply by macosta541** [reply]
Hi Ralf,

Sorry, I emptied the drive because I thought everyone was done with it. Luiz was kind enough to convert the files for me (and all my subsequent data will be acquired by myself on an instrument at the institution I am at), so no need to do further trouble-shooting, I think.  Thank you though!

Best,
Marisa

**Reply by ralfHielscher** [reply]
Rüdiger still had a copy of an old osc file. I think I was able to fix the bug with e6b325ddb3a65194436f863de6ab16759d716a4c.
Are you able to check this?

Ralf.

**Comment by lfgmorales** [comment]
I also trashed the files after I exported them, sorry

Best

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 14 Mar 2021, at 13:42, macosta541 ***@***.***> wrote:

﻿

Hi Ralf,

Sorry, I emptied the drive because I thought everyone was done with it. Luiz was kind enough to convert the files for me (and all my subsequent data will be acquired by myself on an instrument at the institution I am at), so no need to do further trouble-shooting, I think. Thank you though!

Best,
Marisa

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/840#discussioncomment-480379>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN6I3PMRQ2MW3CNFK6LTDSVJHANCNFSM4Y3BUOCA>.

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/840#discussioncomment-480379", "url": "https://github.com/mtex-toolbox/mtex/discussions/840#discussioncomment-480379", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]
